cpp/dolfinx/fem/DofMapBuilder.cpp

```diff
@@ -344,22 +110,51 @@ DofMapStructure build_basic_dofmap(const mesh::Mesh& mesh,
 
   // Compute cell dof permutations
   const Eigen::Array<int, Eigen::Dynamic, Eigen::Dynamic, Eigen::RowMajor>
-      permutations = fem::compute_dof_permutations(mesh, element_dof_layout);
+      permutations
+      = fem::compute_dof_permutations(topology, cell_type, element_dof_layout);
+
+  // Storage for local-to-global map
+  std::vector<std::int64_t> local_to_global(local_size);
+
+  // Dof (dim, entity index) marker
+  std::vector<std::pair<std::int8_t, std::int32_t>> dof_entity(local_size);
 
   // Build dofmaps from ElementDofmap
-  for (auto& cell : mesh::MeshRange(mesh, D, mesh::MeshRangeType::ALL))
+  for (int c = 0; c < connectivity[0]->num_nodes(); ++c)
```

Comment:
Are you starting to remove mesh iterators, in general? In other places I do not see this change.

Reply:
In low-level places it is good to avoid the iterators.

Some problems with the iterators were:
1. They were used in places during mesh construction. This is messy because the Mesh isn't yet fully formed so it's something of a guess as to whether or not the iterators are ready (or possibly only half ready)
2. It was often unclear whether to iterate over all the entities or just the owned entities. Dropping to a lower level forces one to think about this
3. It's easy to make the iterators slow - any changes need careful testing